modules/postman-templating/src/template-client.ts

```diff
@@ -102,47 +67,43 @@ export class TemplateClient {
             )
           }
 
-          // add key regardless, note that this is also returned in lowercase
           variables.add(key)
 
           // if no params continue with the loop
-          if (!params) return
+          if (!params) continue
 
-          // if params provided == attempt to carry out templating
           if (dict[key]) {
             const templated = dict[key]
             tokens.push(templated)
-            return
+          } else {
+            // recipient key must have param
+            if (key === 'recipient') {
+              throw new TemplateError(`Param ${key} not found`)
+            }
           }
-
-          // recipient key must have param
-          if (key === 'recipient')
-            throw new TemplateError(`Param ${templateObject.c} not found`)
-        } else {
-          // FIXME: be more specific about templateObject, just pass the error itself?
-          console.error(
-            `Templating error: invalid template provided. templateObject= ${JSON.stringify(
-              templateObject
-            )}`
-          )
-          throw new TemplateError('Invalid template provided')
+        } else if (tokenType === 'text') {
+          tokens.push(token)
         }
-      })
+      }
+
       return {
         variables: Array.from(variables), // variables are unique
         tokens,
       }
     } catch (err) {
       console.error({ message: `${err.stack}` })
-      if (err.message.includes('unclosed tag'))
+      if (err.message.includes('Unclosed tag'))
         throw new TemplateError(
           'Check that all the keywords have double curly brackets around them.\nA correct example is {{ keyword }}, and incorrect ones are {{ keyword } or {{ keyword . '
         )
-      if (err.message.includes('unclosed string'))
+      // reserved chars in mustache are '^' and '#' and '/'
+      if (
```

Comment:
Try `{{=<%name %>=}}`. This throws "Invalid tags"

Reply:
`{{&name}}` should throw an error - it doesn't

Reply:
{{{name}}} should... throw an error? probably. it doesnt

Reply:
`{{#name}}Never shown!{{/name}}` should also throw an error - it doesn't.

Reply:
`{{! name }}`, `{{> name}}` all pass when they shouldn't. Let's test the cases on https://github.com/janl/mustache.js